sklearn/impute/_base.py

```diff
@@ -612,7 +612,8 @@ def fit(self, X, y=None):
             raise ValueError("'sparse' has to be a boolean or 'auto'. "
                              "Got {!r} instead.".format(self.sparse))
 
-        self.features_ = self._get_missing_features_info(X)[1]
+        self.__missing_features_info = self._get_missing_features_info(X)
```

Comment:
I think we usually only use a single underscore, but otherwise this solution seems good to me.
Do we have a test that checks that ``fit(X).transform(X) == fit_transform(X)``?

Reply:
A single underscore will declare a protected instance variable, which can be accessed outside the class making data transparent, hence I've used double underscore which can't be accessed outside the class which would maintain opacity as the user need not know that data, he is already getting it when using fit_transform.
And moreover I think that can raise issue fit PR checks as the fit function returns self, which would include that protected member, but the tests do not know that fit does return it, so we need to add that to tests.
So, is it okay using private instance variable or should I change it to single?

Reply:
> Do we have a test that checks that `fit(X).transform(X) == fit_transform(X)`?
I don't think so. I'm sorry but I'm very new to test stuff, but I had a look in the test file and don't think that we have something that checks it.

Reply:
could you please add one? it should be quite simple.

Reply:
should I add it to the pre existing test function test_missing_indicator_new

Reply:
okay, I did add a test, I'm pushing it. Please do let me know if any changes are to be made